app/models/linked_profile.rb

```diff
@@ -31,6 +31,35 @@ class LinkedProfile < ApplicationRecord
   belongs_to :user, required: true
   belongs_to :linked_site, required: true
 
+  attr_encrypted :token, key: Base64.encode64(ENV['ATTR_ENCRYPT_KEY'])
+
   validates_presence_of :url, if: :private?
   validates_presence_of :external_user_id
+  validate :verify_mal_credentials
+
+  def verify_mal_credentials
```

Comment:
Make sure you check `sync_to_mal?` before this validation — you should be able to do `validate :verify_mal_credentials, if: :sync_to_mal?`